plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
```

Comment:
True, the TotalAreaCalculator does not need to be a subclass in order to
utilize the functionality of AreaCalculator. For this exercise, in
retrospect, it may be more straightforward not to subclass it. However, in
my mind was the fact that their purpose was essentially the same, with
TotalAreaCalculator being literally an extension of AreaCalculator. If
AreaCalculator were altered in the future (e.g., extended to compute
surface area of 3D objects), or had a constructor, etc., it would make
sense for TotalAreaCalculator to inherit this behavior.

But again, for this exercise (assuming AreaCalculator would never be
changed) I suppose those are considerations aren't pertinent.

On Thu, Feb 5, 2015 at 3:24 PM, Spencer Robinson <notifications@github.com>
wrote:

> In
> plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java
> <https://github.com/siostechcorp/exercises/pull/4#discussion_r24196174>:
>
> > @@ -0,0 +1,25 @@
> > +package planForExtensionExercise;
> > +
> > +public class TotalAreaCalculator extends AreaCalculator {
>
> The TotalAreaCalculator class could get all the functionality of
> AreaCalculator without extending AreaCalculator (it could just reference
> the AreaCalculator.calculateArea method) Can you explain why you chose to
> make the TotalAreaCalculator class a subclass of the AreaCalculator class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siostechcorp/exercises/pull/4/files#r24196174>.
>